datashader/layout.py

```diff
@@ -125,3 +125,67 @@ def __call__(self, nodes, edges, **params):
 
         # Return the nodes with updated positions
         return _merge_points_with_nodes(nodes, points)
+
+
+@nb.jit(nogil=True)
+def _create_random_xy_values(df, seed):
+    np.random.seed(seed)
+
+    points = np.asarray(np.random.random((len(df), 2)))
+    df['x'] = points[:, 0]
+    df['y'] = points[:, 1]
+    return df
+
+
+class random_layout(param.ParameterizedFunction):
+    """
+    Assign coordinates to the nodes randomly.
+    """
+
+    seed = param.Integer(default=None, bounds=(0, 2**32-1), doc="""
+        Random seed used to initialize the pseudo-random number
+        generator.""")
+
+    def __call__(self, nodes, edges, **params):
+        p = param.ParamOverrides(self, params)
+
+        return _create_random_xy_values(nodes.copy(), seed=p.seed)
+
+
+@nb.jit(nogil=True)
+def _create_xy_values_on_unit_circle(df, uniform, seed):
+    np.random.seed(seed)
+
+    r = 0.5  # radius
+    x0, y0 = 0.5, 0.5  # center of unit circle
+    circumference = 2 * np.pi
+
+    if uniform:
+        thetas = np.arange(circumference, step=circumference/len(df))
+    else:
+        thetas = np.asarray(np.random.random((len(df),))) * circumference
+
+    df['x'] = x0 + r * np.cos(thetas)
+    df['y'] = y0 + r * np.sin(thetas)
+
+    return df
```

Comment:
Same here -- if there's no loop, what would Numba have to optimize? Seems like it could be moved into ``circular_layout.__call__``.